### PR TITLE
Skip processing of TXs that we already have.

### PR DESCRIPTION
**Pull request by rebroad**
Previously TXs were processed more than they needed to be even though we already have them.

### CODE REVIEW

File: src/net_processing.cpp [hunk 1]
```diff
@@ -1682,9 +1685,9 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
 
             BOOST_FOREACH(uint256 hash, vEraseQueue)
                 EraseOrphanTx(hash);
-        }
+        } // if accepted to memory pool
         else if (fMissingInputs)
```
Comment (rebroad) [inline src/net_processing.cpp]: for fMissingInputs to be true AlreadyHave() had to be false

File: src/net_processing.cpp [hunk 2]
```diff
@@ -1709,7 +1712,7 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
             } else {
                 LogPrint("mempool", "not keeping orphan with rejected parents %s\n",tx.GetHash().ToString());
             }
-        } else {
+        } else { // not accepted and not missing inputs, i.e. invalid
```
Comment (rebroad) [inline src/net_processing.cpp]: whereas here AlreadyHave() could have been true still

**Comment by gmaxwell** [comment]
&& is shortcutting so this doesn't change it to skip AcceptToMemoryPool.  It does however change it so that it does not run the else clause which is needed for the forced relaying for whitelisted peers (which needs to run already for transactions we already have).

So I think the only real behavior change this makes is to break the functionality.

File: src/net_processing.cpp (hunk 2 above)
Comment (gmaxwell) [inline src/net_processing.cpp]: That is the intended and advertised purpose of the whitelisting, read the comment in the else block!

**Comment by rebroad** [comment]
@gmaxwell Ah, yes, my apologies I has missed the whitelist intention there. It was primarily noticing that `(!tx.HasWitness() && !state.CorruptionPossible())` is bring run against txs we already have in the mempool that made me wonder if there was a logic oversight here, but perhaps the overhead from running this checks again is not significant.

**Pull request closed without merging**